### DISCUSSION TITLE
Is there a way to additionally sign into other providers while in a supabase user session?

### DISCUSSION THREAD

**Discussion by dana-feng** [supabase/supabase#20094] [Questions]
I have a nextjs app that currently uses Supabase's auth.signInWithPassword function; I see that the session has an access token, refresh token etc... However, the goal of my app is to also allow users to connect to other providers such as Google or Github, after they have signed in with Supabase. 

For the google integration, I tried using the implicit flow method: https://developers.google.com/identity/protocols/oauth2/javascript-implicit-flow, but the access token it gives back ends up terminating my existing supabase session and signing the user out of their supabase session after the user signs into their google account. 

Is there a way to merge sessions so that the user can continue to be signed into their supabase account but also get the access tokens from other providers as well? Thank you!

**Comment by GaryAustin1** [comment]
You would likely need to make the OAuth calls to Google  API on your own.  The Supabase auth OAuth is all about getting the user signed in with the provider and clients can only handle 1 method at a time of signin.   It might be possible to have separate clients if you are not using cookies (plain supabase-js) but setting the local storage key so they don't interact.

**Reply by dana-feng** [reply]
Thanks for the reply! I do make the OAuth calls to the google api on my own, by redirecting the user to the oauth page first and then redirecting them back to my app. However, when doing so it appends the google access token as a hash fragment on my app's url, and the supabase session becomes null, even when I store the google access token as a key value pair in local storage. 

I can get around this by storing the supabase session access and refresh tokens in localStorage and then use the supabase.auth.setSession to restore it after the user signs into google, but I don't think that is the best way to go about things.

**Reply by GaryAustin1** [reply]
You can't control where the Google API returns?  Then don't have Supabase in that URL at all.

**Reply by dana-feng** [reply]
I can specify which redirect url to go to after the user authorizes access to google, and I set it to go back to my app, but when it does the user is logged out of the supabase session they previously logged into when they were authenticating into my app

**Reply by GaryAustin1** [reply]
I'll bow out after this one as I've not tried any of this.  
Can you set a supabase client with detectSessionInUrl:false if you come in thru this URL path from the Google API return?

If you are using auth-helpers/ssr though with next.js then not sure if anyway around it, but I'm not that familiar with it.

**Reply by dana-feng** [reply]
Ah yes I am using auth-helpers, but thank you for all your responses!
